storage/tokudb/ha_tokudb.cc

```diff
@@ -3836,41 +3781,16 @@ void ha_tokudb::test_row_packing(uchar* record, DBT* pk_key, DBT* pk_val) {
 }
 
 // set the put flags for the main dictionary
-void ha_tokudb::set_main_dict_put_flags(
-    THD* thd,
-    uint32_t* put_flags) {
-
+void ha_tokudb::set_main_dict_put_flags(THD* thd,
+                                        uint32_t* put_flags) {
     uint32_t old_prelock_flags = 0;
-    uint curr_num_DBs = table->s->keys + tokudb_test(hidden_primary_key);
-    bool in_hot_index = share->num_DBs > curr_num_DBs;
-    bool using_ignore_flag_opt =
-        do_ignore_flag_optimization(
-            thd,
-            table,
-            share->replace_into_fast && !using_ignore_no_key);
-    //
-    // optimization for "REPLACE INTO..." (and "INSERT IGNORE") command
-    // if the command is "REPLACE INTO" and the only table
-    // is the main table (or all indexes are a subset of the pk), 
-    // then we can simply insert the element
-    // with DB_YESOVERWRITE. If the element does not exist,
-    // it will act as a normal insert, and if it does exist, it 
-    // will act as a replace, which is exactly what REPLACE INTO is supposed
-    // to do. We cannot do this if otherwise, because then we lose
-    // consistency between indexes
-    //
+
     if (hidden_primary_key) {
```

Comment:
Merge if (cond) { statement; } else if (another cond) { the same statement; } else if ... to if (cond || another cond) { the same statement} else if ...